### ISSUE TITLE
Setting raygun up with asp.net core 2.0

### ISSUE THREAD

**Issue by xt0rted** [MindscapeHQ/raygun4net#367]
I'm migrating a project of mine from classic asp.net & raygun.owin to asp.net core 2.0 and the official client. In 2.0 they've streamlined a lot of the boiler plate so the config instance you work with is an `IConfiguration` not an `IConfigurationRoot`. This means the out of the box `services.AddRaygun(Configuration)` extension method no longer works with the recommended startup configuraiton. This is talked about in the 1.x to 2.0 migration docs https://docs.microsoft.com/en-us/aspnet/core/migration/1x-to-2x/#add-configuration-providers.

Given this core setup change should raygun be updated to take in an `IConfiguration` now?  This is the default `Startup.cs` when you do File > New Project

```csharp
namespace EquineExchange.Homepage
{
    using Microsoft.AspNetCore.Builder;
    using Microsoft.AspNetCore.Hosting;
    using Microsoft.Extensions.Configuration;
    using Microsoft.Extensions.DependencyInjection;

    using Mindscape.Raygun4Net;

    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddMvc();
            services.AddRaygun(Configuration); // gives an error saying IConfiguration can't be assigned to IConfigurationRoot
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseBrowserLink();
            }
            else
            {
                app.UseExceptionHandler("/Home/Error");
            }

            app.UseRaygun();

            app.UseStaticFiles();

            app.UseMvc(routes =>
            {
                routes.MapRoute(
                    name: "default",
                    template: "{controller=Home}/{action=Index}/{id?}");
            });
        }
    }
}
```

For now I'm wiring everything up manually like so
```csharp
services.Configure<RaygunSettings>(Configuration.GetSection("RaygunSettings"));
services.AddTransient<IRaygunAspNetCoreClientProvider>(_ => new DefaultRaygunAspNetCoreClientProvider());
services.AddSingleton<RaygunMiddlewareSettings>();
```

**Comment by xt0rted** [comment]
I originally wrote [Raygun.Owin](https://github.com/xt0rted/Raygun.Owin) for the same reason. Was hoping to not have to do it again for .net core but I'll have to give this a try and see how it works. Thanks for doing this and letting us know about it.

**Comment by mduncan26** [comment]
Hi @xt0rted 

Just letting you know that we have officially released a new provider to support .Net Core 2 and the .Net Standard 2. It can be found on NuGet here: https://www.nuget.org/packages/Mindscape.Raygun4Net.NetCore/ and the source is located here in our repository: https://github.com/MindscapeHQ/raygun4net/tree/master/Mindscape.Raygun4Net.NetCore.

**Comment by xt0rted** [comment]
@mduncan26 shouldn't the [asp.net core project](https://github.com/MindscapeHQ/raygun4net/tree/master/Mindscape.Raygun4Net.AspNetCore) be updated to depend on this new package then? That's the one I, and others, are currently using with .net core.

The only issue I had with that package is the extension method to wire it up because of how asp.net core 2 modified how you setup your configs. In the end this wasn't an issue because the extension overload takes an `IConfiguration` and I needed to customize the client provider which is done via this version of the method.

Currently I'm using the asp.net core provider like so:

```csharp
services.AddRaygun(Configuration, new RaygunMiddlewareSettings
{
    ClientProvider = new VersionedRaygunClientProvider()
});
```

```csharp
public class VersionedRaygunClientProvider : DefaultRaygunAspNetCoreClientProvider
{
    public override RaygunClient GetClient(RaygunSettings settings, HttpContext context)
    {
        var client = base.GetClient(settings, context);

        client.ApplicationVersion = ApplicationInformation.Version;

        return client;
    }
}
```

```csharp
public static class ApplicationInformation
{
    static ApplicationInformation()
    {
        var assembly = typeof(ApplicationInformation).Assembly;

        Version = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
    }

    public static string Version { get; }
}
```

**Comment by mduncan26** [comment]
@xt0rted Yes we are planning to have the ASP .NET Core provider use this new provider as its base. That work will be taken up in the future when we have the resources available. I will make a note to address the configuration issues as well.

**Comment by xt0rted** [comment]
This was fixed in #386
